Alfie/Alfie/Views/ProductDetails/ProductDetailsViewModel.swift

```diff
@@ -87,16 +88,46 @@ final class ProductDetailsViewModel: ProductDetailsViewModelProtocol {
         product?.priceType
     }
 
-    init(productId: String, product: Product?, dependencies: ProductDetailsDependencyContainer) {
+    var isAddToBagEnabled: Bool {
+        productHasStock
+        && hasColorSelected
+        && hasSizeSelected
+    }
+
+    private var hasColorSelected: Bool {
+        colorSelectionConfiguration.items.count > 1 ? selectedVariant?.colour != nil : true
+    }
+
+    private var hasSizeSelected: Bool {
+        sizingSelectionConfiguration.items.count > 1 ? selectedVariant?.size != nil : true
```

Comment:
Same as the `hasColorSelected`:

```suggestion
        !colorSelection.configuration.items.isEmpty && selectedVariant?.size != nil
```